backend/siarnaq/api/episodes/models.py

```diff
@@ -270,12 +283,90 @@ class Tournament(models.Model):
     def __str__(self):
         return self.name_short
 
-    def seed_by_scrimmage(self):
+    def get_potential_participants(self):
+        """Returns the list of participants that would be entered in this tournament,
+        if it were to start right now."""
+        # TODO incorporate real eligibility stuff,
+        # rebasing and using #545
+        return (
+            Team.objects.with_active_submission()
+            .filter(episode=self.episode)
+            .all()
+            .order_by("-profile__rating__value")
+        )
+
+    def initialize(self):
         """
-        Seed the tournament with eligible teamsn in order of decreasing rating, and
-        populate the Challonge brackets.
+        Seed the tournament with eligible teams in order of decreasing rating,
+        populate the Challonge brackets, and create TournamentRounds.
         """
-        raise NotImplementedError
+
+        tournament_name_public = f"{self.episode.name_long} {self.name_long}"
+        tournament_name_private = tournament_name_public + " (private)"
+
+        # For security by obfuscation,
+        # and to allow easy regeneration of bracket
+        key = random.randint(1000, 9999)
+        # Challonge does not allow hyphens in its IDs
+        # so substitute them just in case
+        tournament_id_public = f"{self.name_short}_{key}".replace("-", "_")
+        tournament_id_private = f"{tournament_id_public}_private"
+
+        # NOTE: We don't support double elim yet.
+        # Tracked in #548. (Also make sure to actually read the "style" field)
+        is_single_elim = True
```

Comment:
Don't like this. Why can't the function just take a `TournamentStyle` parameter?